rust/stackable-cockpit/src/platform/service.rs

```diff
@@ -45,15 +45,54 @@ pub async fn get_endpoints(
     object_name: &str,
     object_namespace: &str,
 ) -> Result<IndexMap<String, String>, Error> {
-    let service_list_params =
+    let mut endpoints = IndexMap::new();
+    let list_params =
         ListParams::from_product(product_name, Some(object_name), k8s::ProductLabel::Name);
 
-    let services = kube_client
-        .list_services(Some(object_namespace), &service_list_params)
+    let listeners = kube_client
+        .list_listeners(Some(object_namespace), &list_params)
         .await
         .context(KubeClientFetchSnafu)?;
 
-    let mut endpoints = IndexMap::new();
+    for listener in &listeners {
+        let Some(listener_status) = &listener.status else {
+            continue;
+        };
+        for address in listener_status.ingress_addresses.iter().flatten() {
+            for port in &address.ports {
+                // Listener names usually have the pattern "listener-simple-hdfs-namenode-default-0" or
+                // "simple-hdfs-datanode-default-0-listener", so we can strip everything before the first occurrence of
+                // the stacklet name ("simple-hdfs" in this case). After that it actually get's pretty hard.
+                // This truncation is *not* ideal, however we only have implemented listener-operator for HDFS so far,
+                // so better to have support for that than nothing :)
+                let listener_name = listener.name_any();
+                let Some((_, display_name)) = listener_name.split_once(object_name) else {
+                    continue;
+                };
+                let text = format!(
+                    "{display_name}-{port_name}",
+                    display_name = display_name.trim_start_matches('-'),
+                    port_name = port.0
+                );
+                let endpoint_url = endpoint_url(&address.address, *port.1, port.0);
+                endpoints.insert(text, endpoint_url);
+            }
+        }
+    }
+
+    // Ideally we use listener-operator everywhere, afterwards we can remove the whole k8s Services handling following.
+    // Currently the Services created by listener-op are missing the recommended labels, so this early exit in case we
+    // find Listeners is currently not required. However, once we add the recommended labels to the k8s Services, we
+    // would have duplicated entries (one from the Listener and one from the Service). Because of this we don't look at
+    // the Services in case we found Listeners!
```

Comment:
suggestion: Slight text adjustment

```suggestion
    // Ideally we use listener-operator everywhere, afterwards we can remove the whole k8s Services handling below.
    // Currently the Services created by listener-op are missing the recommended labels, so this early exit in case we
    // find Listeners is currently not required. However, once we add the recommended labels to the k8s Services, we
    // would have duplicated entries (one from the Listener and one from the Service). Because of this we don't look at
    // the Services in case we found Listeners!
```